SimG4CMS/Calo/src/CaloSD.cc

```diff
@@ -341,8 +341,7 @@ bool CaloSD::checkHit() {
 
 int CaloSD::getNumberOfHits() { return theHC->entries(); }
 
-CaloG4Hit* CaloSD::createNewHit(const G4Step* aStep) {
-  auto const theTrack = aStep->GetTrack();
+CaloG4Hit* CaloSD::createNewHit(const G4Step* aStep, const G4Track* theTrack) {
```

Comment:
@civanch I added theTrack parameter here because in case of GFlash hits aStep->GetTrack() will return nullptr and there is no way except const_cast to set it.